opensrp-opd/src/main/java/org/smartregister/opd/utils/LookUpUtils.java

```diff
@@ -0,0 +1,135 @@
+//                        if (progressBar != null) {
+//                            progressBar.setVisibility(VISIBLE);
+//                        }
+                    }
+                }, null);
+    }
+    private static List<CommonPersonObject> clientLookUp(Context context, Map<String, String> entityLookUp) {
+        List<CommonPersonObject> results = new ArrayList<>();
+        if (context == null) {
+            return results;
+        }
+
+
+        if (entityLookUp.isEmpty()) {
+            return results;
+        }
+
+        String tableName = Utils.metadata().getTableName();
+
+        CommonRepository commonRepository = context.commonrepository(tableName);
+        String query = lookUpQuery(entityLookUp, tableName);
+
+        Cursor cursor = null;
+        try {
+
+            cursor = commonRepository.rawCustomQueryForAdapter(query);
+            if (cursor != null && cursor.getCount() > 0 && cursor.moveToFirst()) {
+                while (!cursor.isAfterLast()) {
+                    CommonPersonObject commonPersonObject = commonRepository.readAllcommonforCursorAdapter(cursor);
+                    results.add(commonPersonObject);
+                    cursor.moveToNext();
+                }
+            }
+
+
+        } catch (Exception e) {
+            Timber.e(e);
+        } finally {
+            if (cursor != null) {
+                cursor.close();
+            }
+        }
+
+        return results;
+    }
+
+    private static String lookUpQuery(Map<String, String> entityMap, String tableName) {
+        SmartRegisterQueryBuilder queryBUilder = new SmartRegisterQueryBuilder();
+        queryBUilder.SelectInitiateMainTable(tableName,
+                new String[]{Constants.KEY.RELATIONALID,Constants.KEY.MER_ID, Constants.KEY.FIRST_NAME, Constants.KEY.LAST_NAME,
+                        AllConstants.ChildRegistrationFields.GENDER, Constants.KEY.DOB,
+                        Constants.KEY.BASE_ENTITY_ID}
+
+        );
+        String query = queryBUilder.mainCondition(getMainConditionString(entityMap));
+        return queryBUilder.Endquery(query);
+    }
+
+    private static String getMainConditionString(Map<String, String> entityMap) {
+        String mainConditionString = "";
+        for (Map.Entry<String, String> entry : entityMap.entrySet()) {
+            String key = entry.getKey();
+            String value = entry.getValue();
+            //first name, last name, bht id, national id
+            String firstName = "first_name";
+            String lastName = "last_name";
+            String bht_id = "bht_mid";
+            String national_id = "national_id";
+            if (StringUtils.containsIgnoreCase(key, firstName)) {
+                key = firstName;
+            }
+
+            if (StringUtils.containsIgnoreCase(key, lastName)) {
+                key = lastName;
+            }
+
+            if (StringUtils.containsIgnoreCase(key, bht_id)) {
```

Comment:
Should this be an `else if`? Not sure why it checks for `contains`

Reply:
its sort of aggregates all the conditions needed